cli/src/main.rs

```diff
@@ -95,44 +98,29 @@ pub fn parse_args(matches: &ArgMatches<'_>) -> Result<CliConfig, Box<dyn error::
         default.json_rpc_url
     };
 
-    let CliCommandInfo {
-        command,
-        require_keypair,
-    } = parse_command(&matches)?;
-
-    let (keypair, keypair_path) = if require_keypair {
-        let path = if matches.is_present("keypair") {
-            matches.value_of("keypair").unwrap().to_string()
-        } else if config.keypair_path != "" {
-            config.keypair_path
-        } else {
-            let default_keypair_path = CliConfig::default_keypair_path();
-            if !std::path::Path::new(&default_keypair_path).exists() {
-                return Err(CliError::KeypairFileNotFound(format!(
-                    "Generate a new keypair at {} with `solana-keygen new`",
-                    default_keypair_path
-                ))
-                .into());
-            }
-            default_keypair_path
-        };
-
-        let keypair = keypair_util_from_path(matches, &path, "keypair")?;
-        (keypair, Some(path))
+    let default_signer_path = if matches.is_present("keypair") {
+        matches.value_of("keypair").unwrap().to_string()
+    } else if config.keypair_path != "" {
+        config.keypair_path
     } else {
-        let default = CliConfig::default();
-        (default.keypair, None)
+        CliConfig::default_keypair_path()
     };
 
-    Ok(CliConfig {
-        command,
-        json_rpc_url,
-        keypair,
-        keypair_path,
-        derivation_path: derivation_of(matches, "derivation_path"),
-        rpc_client: None,
-        verbose: matches.is_present("verbose"),
-    })
+    let CliCommandInfo { command, signers } =
+        parse_command(&matches, &default_signer_path, &wallet_manager)?;
+
+    Ok((
+        CliConfig {
+            command,
+            json_rpc_url,
+            signers: vec![],
+            keypair_path: Some(default_signer_path), // TODO: this isn't correct when default signer not used
```

Comment:
Is this TODO intended to be resolved here or in the future?  Might be good to surface as a Github issue and reference it here, if the latter

Reply:
Ah, thanks, meant to come back to this.
I opted to remove the option and specify that signer as "default". I think we could do a better job of verbosely reporting which signers are signing various subcommand transactions, but that can be the work of another PR.